docs/_tutorials/automatic-tensor-parallelism.md

```diff
@@ -0,0 +1,129 @@
+---
+title: "Automatic Tensor Parallelism for HuggingFace Models"
+tags: inference
+---
+
+This tutorial demonstrates the new automatic tensor parallelism feature for inference. Previously, the user needed to provide an injection policy to DeepSpeed to enable tensor parallelism. DeepSpeed now supports automatic tensor parallelism for HuggingFace models by default as long as kernel injection is not true and an injection policy is not provided. This is convenient for when the injection policy of a model is not known and for improving performance of models without kernel injection support.
+
+```python
+# create the model
+import transformers
+import deepspeed
+pipe = pipeline("text2text-generation", model="google/t5-v1_1-small", device=local_rank)
+# Initialize the DeepSpeed-Inference engine
+pipe.model = deepspeed.init_inference(
+    pipe.model,
+    mp_size=world_size,
+    dtype=torch.float
+)
+output = pipe('Input String')
+```
```

Comment:
```suggestion
import os
import torch
import transformers
import deepspeed

local_rank = int(os.getenv("LOCAL_RANK", "0"))
world_size = int(os.getenv("WORLD_SIZE", "1"))

# create the model pipeline
pipe = transformers.pipeline(task="text2text-generation", model="google/t5-v1_1-small", device=local_rank)
# Initialize the DeepSpeed-Inference engine
pipe.model = deepspeed.init_inference(
    pipe.model,
    mp_size=world_size,
    dtype=torch.float
)
output = pipe('Input String')
```

We should probably make the example code runnable if a user were to copy, paste, and run it.

Reply:
That's a good point. I referenced the same example from the inference tutorial page. I had assumed it was runnable myself.